github/GithubIntegration.py

```diff
@@ -0,0 +1,140 @@
+import github.PaginatedList
+import github.Requester
+
+from . import Consts, GithubException
+
+
+class GithubIntegration:
+    """
+    Main class to obtain tokens for a GitHub integration.
+    """
+
+    def __init__(self, integration_id, private_key, base_url=Consts.DEFAULT_BASE_URL):
+        """
+        :param base_url: string
+        :param integration_id: int
+        :param private_key: string
+        """
+        self.base_url = base_url
+        self.integration_id = integration_id
+        self.private_key = private_key
+        assert isinstance(base_url, str), base_url
+
+    def create_jwt(self, expiration=60):
+        """
+        Creates a signed JWT, valid for 60 seconds by default.
+        The expiration can be extended beyond this, to a maximum of 600 seconds.
+
+        :param expiration: int
+        :return string:
+        """
+        now = int(time.time())
+        payload = {"iat": now, "exp": now + expiration, "iss": self.integration_id}
+        encrypted = jwt.encode(payload, key=self.private_key, algorithm="RS256")
+
+        if isinstance(encrypted, bytes):
+            encrypted = encrypted.decode("utf-8")
+
+        return encrypted
+
+    def get_access_token(self, installation_id, user_id=None):
+        """
+        Get an access token for the given installation id.
+        POSTs https://api.github.com/app/installations/<installation_id>/access_tokens
+        :param user_id: int
+        :param installation_id: int
+        :return: :class:`github.InstallationAuthorization.InstallationAuthorization`
+        """
+        body = {}
+        if user_id:
+            body = {"user_id": user_id}
+        response = requests.post(
+            f"{self.base_url}/app/installations/{installation_id}/access_tokens",
+            headers={
+                "Authorization": f"Bearer {self.create_jwt()}",
+                "Accept": Consts.mediaTypeIntegrationPreview,
+                "User-Agent": "PyGithub/Python",
+            },
+            json=body,
+        )
+
+        if response.status_code == 201:
+            return github.InstallationAuthorization.InstallationAuthorization(
+                requester=None,  # not required, this is a NonCompletableGithubObject
+                headers={},  # not required, this is a NonCompletableGithubObject
+                attributes=response.json(),
+                completed=True,
+            )
+        elif response.status_code == 403:
+            raise GithubException.BadCredentialsException(
+                status=response.status_code, data=response.text
+            )
+        elif response.status_code == 404:
+            raise GithubException.UnknownObjectException(
+                status=response.status_code, data=response.text
+            )
+        raise GithubException.GithubException(
+            status=response.status_code, data=response.text
+        )
+
+    @deprecated.deprecated("Use get_repo_installation")
```

Comment:
Let me know if that does not suit you. I feel that this call is badly named

Reply:
This is the existing call? It should probably also state it's deprecated in the docstring. Also, it should mention the class, not just the bare function.